### PR TITLE
fix(ocm): add data_server_url to ocm storage provider

### LINKED ISSUE #10358: OCM cannot open shared file
## Describe the bug
When I share a file using OCM, I cannot open it as the share receiver.

## Steps to reproduce
1. use the OCM deployment example from https://github.com/owncloud/ocis-charts/pull/701/
2. share a text file from oCIS 1 to oCIS 2
3. open the text file as share receiver

## Expected behavior

The text file can be opened and edited.

## Actual behavior

https://ocis.kube.owncloud.test/remote.php/dav/spaces/89f37a33-858b-45fa-8890-a1f2b27d90e1%24c04eee26-d9b9-4238-9b51-4aeb3b9f330b!Lw%3D%3D fails with HTTP 500

```
proxy-65d7c5746f-qp4nj proxy {"level":"error","service":"proxy","error":"failed to verify access token: token is malformed: token contains an invalid number of segments","authenticator":"oidc","path":"/dav/ocm/5caffad9-acff-480b-a370-5e3c1e0e87cb/","user_agent":"Go-http-client/1.1","client.address":"10.244.0.1","network.peer.address":"","network.peer.port":"","time":"2024-10-21T09:37:47Z","message":"failed to authenticate the request"}
proxy-65d7c5746f-qp4nj proxy {"level":"error","service":"proxy","error":"failed to verify access token: token is malformed: token contains an invalid number of segments","authenticator":"oidc","path":"/dav/ocm/5caffad9-acff-480b-a370-5e3c1e0e87cb/","user_agent":"Go-http-client/1.1","client.address":"10.244.0.1","network.peer.address":"","network.peer.port":"","time":"2024-10-21T09:37:47Z","message":"failed to authenticate the request"}
```

## Setup

oCIS 6.5 deployed with https://github.com/owncloud/ocis-charts/pull/701/

### PR DESCRIPTION
**Pull request by rhafer**
Fixes #10358

### CODE REVIEW

File: services/ocm/pkg/config/config.go [hunk 1]
```diff
@@ -121,8 +121,9 @@ type OCMCore struct {
 	Drivers OCMCoreDrivers `yaml:"drivers"`
 }
 type OCMStorageProvider struct {
-	Insecure    bool   `yaml:"insecure" env:"OCM_OCM_STORAGE_PROVIDER_INSECURE" desc:"Disable TLS certificate validation for the OCM connections. Do not set this in production environments." introductionVersion:"5.0"`
-	StorageRoot string `yaml:"storage_root" env:"OCM_OCM_STORAGE_PROVIDER_STORAGE_ROOT" desc:"Directory where the ocm storage provider persists its data like tus upload info files." introductionVersion:"5.0"`
+	Insecure      bool   `yaml:"insecure" env:"OCM_OCM_STORAGE_PROVIDER_INSECURE" desc:"Disable TLS certificate validation for the OCM connections. Do not set this in production environments." introductionVersion:"5.0"`
+	StorageRoot   string `yaml:"storage_root" env:"OCM_OCM_STORAGE_PROVIDER_STORAGE_ROOT" desc:"Directory where the ocm storage provider persists its data like tus upload info files." introductionVersion:"5.0"`
+	DataServerURL string `yaml:"data_server_url" env:"OCM_OCM_STORAGE_DATA_SERVER_URL" desc:"URL of the data server, needs to be reachable by the data gateway provided by the frontend service or the user if directly exposed." introductionVersion:"7.0.0"`
```
Comment (mmattel) [inline services/ocm/pkg/config/config.go:126]: This is very hard to understand what it is good for. Please add a text in the readme to explain in more detail.

Comment (rhafer) [reply services/ocm/pkg/config/config.go:126]: The text is a copy from https://github.com/owncloud/ocis/blob/master/services/storage-users/pkg/config/config.go#L29 :laughing:  I have a hard time coming up with something better that is brief enough to fit into a comment here. I'll try.

Comment (wkloucek) [reply services/ocm/pkg/config/config.go:126]: I must agree to @rhafer that it's a very precise and understandable description, if you know what:
- a data gateway is 
- a data server is

But this is oCIS architecture fundamentals that can not be covered in a config option description

**Review by wkloucek — APPROVED** [review]
Looks fine from a configuration perspective

File: services/ocm/pkg/config/config.go (hunk 1 above)
Comment (rhafer) [reply services/ocm/pkg/config/config.go:126]: > I proposed to do a brief explanation in the readme to help admins to understand

I agree that a better description of the architecture is needed here. (I am not even sure I understand it well enough to properly describe it).
But this PR is fixing a release blocker. So I think we should merge it independently of the poor documentation. That issue is broader anyway it affects every storageprovider.

**Pull request merged**